sklearn/ensemble/_bagging.py

```diff
@@ -120,10 +135,12 @@ def _parallel_build_estimators(
                 not_indices_mask = ~indices_to_mask(indices, n_samples)
                 curr_sample_weight[not_indices_mask] = 0
 
-            estimator.fit(X[:, features], y, sample_weight=curr_sample_weight)
+            # only index arrays when needed
+            X_ = X if max_features == X.shape[1] else X[:, features]
```

Comment:
I would do this in a separate follow up PR. The `check_input` change is already an improvement that can be merged by itself.

As for this change, we can only do this optimization if `bootstrap_features` is False. `bootstrap_features` is always False for `IsolationForest`, but not for `Bagging*`. (If `bootstrap_features` is True, then the features are sampled with replacement which can result in repeated features.)

Reply:
Got it! I've reverted the changes in this PR, and will draft another PR for this one